### PR TITLE
Add support for uClibc

### PR DESCRIPTION
**Pull request by WorksButNotTested**

### CODE REVIEW

File: gio/ginetaddress.c [hunk 1]
```diff
@@ -106,10 +114,18 @@ g_inet_address_set_property (GObject      *object,
...
       memcpy (&address->priv->addr, g_value_get_pointer (value),
 	      address->priv->family == AF_INET ?
 	      sizeof (address->priv->addr.ipv4) :
 	      sizeof (address->priv->addr.ipv6));
+#else
+      if (address->priv->family == AF_INET)
+      {
+        memcpy (&address->priv->addr, g_value_get_pointer (value),
+	        sizeof (address->priv->addr.ipv4));
+      }
```
Comment (oleavr) [inline gio/ginetaddress.c]: - The whole block should be indented by 2 additional spaces.
- The hanging indent should be 4, not 6 spaces.

File: gio/ginetaddress.c [hunk 2]
```diff
@@ -521,9 +540,15 @@ g_inet_address_new_from_string (const gchar *string)
   g_networking_init ();
 
   if (inet_pton (AF_INET, string, &in_addr) > 0)
+  {
     return g_inet_address_new_from_bytes ((guint8 *)&in_addr, AF_INET);
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Let's not add this block, want to keep changes on top of upstream code to a minimum.

File: gio/ginetaddress.c [hunk 3]
```diff
@@ -521,9 +540,15 @@ g_inet_address_new_from_string (const gchar *string)
...
+#ifdef HAVE_IPV6
   else if (inet_pton (AF_INET6, string, &in6_addr) > 0)
+  {
     return g_inet_address_new_from_bytes ((guint8 *)&in6_addr, AF_INET6);
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Let's not add this block, want to keep changes on top of upstream code to a minimum.

File: gio/ginetaddress.c [hunk 4]
```diff
@@ -510,7 +526,10 @@ GInetAddress *
 g_inet_address_new_from_string (const gchar *string)
 {
   struct in_addr in_addr;
+
```
Comment (oleavr) [inline gio/ginetaddress.c]:
```suggestion
```

File: gio/ginetaddress.c [hunk 5]
```diff
@@ -626,16 +663,25 @@ g_inet_address_new_any (GSocketFamily family)
 gchar *
 g_inet_address_to_string (GInetAddress *address)
 {
+
```
Comment (oleavr) [inline gio/ginetaddress.c]:
```suggestion
```

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks! I've highlighted a few coding style issues that apply to most of this PR:

File: gio/ginetaddress.c (hunk 1 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: Apologies. It seems that this file (and many others from glib) incorporate both tabs and spaces for their indentation! I assume that frida inherited this from the parent project. So, this looked just fine on my machine!

I replaced the tabs with spaces for the section which I changed. But didn't want to touch anything else as it would make large sections of the project different from the upstream and hence difficult to pull in updates.

Hope that's ok?

File: gio/ginetaddress.c (hunk 2 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: Sure. This one and the next weren't needed. I checked through the rest and they all incorporated the following structure:

else
{
#ifdef HAVE_IPV6
  do something
#else
  do something else
#endif
}

These technically could be reduced to a single line if, as the pre-processor would reduce the code before compilation, however although the 'contributing' page didn't explicitly say what I should do, I thought that this may make the code hard to read so I introduced a braced block where a single-line else clause had been used before.

Subsequently, the 'contributing' page said to use a braced block for both parts of the if/else if either uses them and so I introduced braces on the if clause too match. Again, I am not sure whether you agree this is the right approach as it isn't so clear cut when you are working with third-party code how much you want to retrofit your own coding style.

Happy to change things either way, just let me know what works best.

File: gio/ginetaddress.c (hunk 3 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: See previous.

File: gio/ginetaddress.c (hunk 4 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: Deleted

File: gio/ginetaddress.c (hunk 5 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: Deleted

File: gio/ginetaddress.c [hunk 6]
```diff
@@ -93,9 +101,9 @@ enum
 
 static void
 g_inet_address_set_property (GObject      *object,
-			     guint         prop_id,
-			     const GValue *value,
-			     GParamSpec   *pspec)
+                             guint         prop_id,
+                             const GValue *value,
+                             GParamSpec   *pspec)
```
Comment (oleavr) [inline gio/ginetaddress.c]: Let's avoid making unnecessary changes, as it will make future patch rebase operations heavier.

File: gio/ginetaddress.c [hunk 7]
```diff
@@ -106,10 +114,18 @@ g_inet_address_set_property (GObject      *object,
...
 
     case PROP_BYTES:
+#ifdef HAVE_IPV6
       memcpy (&address->priv->addr, g_value_get_pointer (value),
-	      address->priv->family == AF_INET ?
-	      sizeof (address->priv->addr.ipv4) :
-	      sizeof (address->priv->addr.ipv6));
+          address->priv->family == AF_INET ?
+          sizeof (address->priv->addr.ipv4) :
+          sizeof (address->priv->addr.ipv6));
```
Comment (oleavr) [inline gio/ginetaddress.c]: This also looks like an unnecessary change and should be dropped.

File: gio/ginetaddress.c [hunk 8]
```diff
@@ -106,10 +114,18 @@ g_inet_address_set_property (GObject      *object,
...
+#else
+      if (address->priv->family == AF_INET)
+      {
+        memcpy (&address->priv->addr, g_value_get_pointer (value),
+            sizeof (address->priv->addr.ipv4));
+      }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 9]
```diff
@@ -574,13 +594,19 @@ g_inet_address_new_loopback (GSocketFamily family)
...
   if (family == AF_INET)
-    {    
-      guint8 addr[4] = {127, 0, 0, 1};
+  {
+    guint8 addr[4] = {127, 0, 0, 1};
 
-      return g_inet_address_new_from_bytes (addr, family);
-    }
+    return g_inet_address_new_from_bytes (addr, family);
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 10]
```diff
@@ -574,13 +594,19 @@ g_inet_address_new_loopback (GSocketFamily family)
...
   else
+  {
+#ifdef HAVE_IPV6
     return g_inet_address_new_from_bytes (in6addr_loopback.s6_addr, family);
+#else
+    return NULL;
+#endif
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 11]
```diff
@@ -602,13 +628,19 @@ g_inet_address_new_any (GSocketFamily family)
...
   if (family == AF_INET)
-    {    
-      guint8 addr[4] = {0, 0, 0, 0};
+  {
+    guint8 addr[4] = {0, 0, 0, 0};
 
-      return g_inet_address_new_from_bytes (addr, family);
-    }
+    return g_inet_address_new_from_bytes (addr, family);
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 12]
```diff
@@ -602,13 +628,19 @@ g_inet_address_new_any (GSocketFamily family)
...
   else
+  {
+#ifdef HAVE_IPV6
     return g_inet_address_new_from_bytes (in6addr_any.s6_addr, family);
+#else
+    return NULL;
+#endif
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 13]
```diff
@@ -631,11 +663,19 @@ g_inet_address_to_string (GInetAddress *address)
   g_return_val_if_fail (G_IS_INET_ADDRESS (address), NULL);
 
   if (address->priv->family == AF_INET)
+  {
     inet_ntop (AF_INET, &address->priv->addr.ipv4, buffer, sizeof (buffer));
+    return g_strdup (buffer);
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 14]
```diff
@@ -631,11 +663,19 @@ g_inet_address_to_string (GInetAddress *address)
...
+  {
+#ifdef HAVE_IPV6
     inet_ntop (AF_INET6, &address->priv->addr.ipv6, buffer, sizeof (buffer));
-
-  return g_strdup (buffer);
+    return g_strdup (buffer);
+#else
+    return NULL;
+#endif
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 15]
```diff
@@ -711,13 +755,19 @@ g_inet_address_get_is_any (GInetAddress *address)
...
   if (address->priv->family == AF_INET)
-    {
-      guint32 addr4 = g_ntohl (address->priv->addr.ipv4.s_addr);
+  {
+    guint32 addr4 = g_ntohl (address->priv->addr.ipv4.s_addr);
 
-      return addr4 == INADDR_ANY;
-    }
+    return addr4 == INADDR_ANY;
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

File: gio/ginetaddress.c [hunk 16]
```diff
@@ -711,13 +755,19 @@ g_inet_address_get_is_any (GInetAddress *address)
...
   else
+  {
+#ifdef HAVE_IPV6
     return IN6_IS_ADDR_UNSPECIFIED (&address->priv->addr.ipv6);
+#else
+    return FALSE;
+#endif
+  }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block indentation is off.

Comment (oleavr) [reply gio/ginetaddress.c]: (Would be good to avoid adding blocks when not needed though, to make the patch smaller.)

**Review by oleavr — CHANGES_REQUESTED** [review]

**Comment by WorksButNotTested** [comment]
Fixed indentation of braced if blocks

File: gio/ginetaddress.c [hunk 17]
```diff
@@ -574,13 +594,19 @@ g_inet_address_new_loopback (GSocketFamily family)
   g_return_val_if_fail (G_INET_ADDRESS_FAMILY_IS_VALID (family), NULL);
 
   if (family == AF_INET)
-    {    
+    {
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change. (Even if technically an improvement, it makes the patch larger.)

File: gio/ginetaddress.c [hunk 18]
```diff
@@ -602,13 +628,19 @@ g_inet_address_new_any (GSocketFamily family)
...
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change. (Even if technically an improvement, it makes the patch larger.)

File: gio/ginetaddress.c [hunk 19]
```diff
@@ -602,13 +628,19 @@ g_inet_address_new_any (GSocketFamily family)
...
       guint8 addr[4] = {0, 0, 0, 0};
 
       return g_inet_address_new_from_bytes (addr, family);
     }
   else
-    return g_inet_address_new_from_bytes (in6addr_any.s6_addr, family);
+    {
+#ifdef HAVE_IPV6
+     return g_inet_address_new_from_bytes (in6addr_any.s6_addr, family);
```
Comment (oleavr) [inline gio/ginetaddress.c]: Indentation off by one space.

File: gio/ginetaddress.c (hunk 16 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: Should I remove the block on line 764? It looks like this

    {
#ifdef HAVE_IPV6
      return IN6_IS_ADDR_UNSPECIFIED (&address->priv->addr.ipv6);
#else
      return FALSE;
#endif
    }

I thought we wanted to keep the blocks unless the statement was only a single line. Although the pre-processor will reduce this to a single line, should the braces be left here to make it easier to read?

File: gio/ginetaddress.c (hunk 19 above)
Comment (WorksButNotTested) [reply gio/ginetaddress.c]: Sorry. Having a bit of a nightmare with my editor (vscode), seems it really isn't getting on too well with glib. I have found a fixed a few others too.

https://github.com/frida/glib/pull/6/commits/a31667a6ad4d516a684e16b2375d7aa3fc15fb2d

File: gio/ginetaddress.c [hunk 20]
```diff
@@ -580,7 +600,13 @@ g_inet_address_new_loopback (GSocketFamily family)
...
     }
   else
-    return g_inet_address_new_from_bytes (in6addr_loopback.s6_addr, family);
+    {
+#ifdef HAVE_IPV6
+      return g_inet_address_new_from_bytes (in6addr_loopback.s6_addr, family);
+#else
+      return NULL;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 21]
```diff
@@ -608,7 +634,13 @@ g_inet_address_new_any (GSocketFamily family)
...
     }
   else
-    return g_inet_address_new_from_bytes (in6addr_any.s6_addr, family);
+    {
+#ifdef HAVE_IPV6
+      return g_inet_address_new_from_bytes (in6addr_any.s6_addr, family);
+#else
+      return NULL;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 22]
```diff
@@ -717,7 +761,13 @@ g_inet_address_get_is_any (GInetAddress *address)
...
     }
   else
-    return IN6_IS_ADDR_UNSPECIFIED (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_UNSPECIFIED (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 23]
```diff
@@ -743,7 +793,13 @@ g_inet_address_get_is_loopback (GInetAddress *address)
...
     }
   else
-    return IN6_IS_ADDR_LOOPBACK (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_LOOPBACK (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 24]
```diff
@@ -771,7 +827,13 @@ g_inet_address_get_is_link_local (GInetAddress *address)
...
     }
   else
-    return IN6_IS_ADDR_LINKLOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_LINKLOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 25]
```diff
@@ -802,7 +864,13 @@ g_inet_address_get_is_site_local (GInetAddress *address)
...
     }
   else
-    return IN6_IS_ADDR_SITELOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_SITELOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 26]
```diff
@@ -827,7 +895,13 @@ g_inet_address_get_is_multicast (GInetAddress *address)
...
     }
   else
-    return IN6_IS_ADDR_MULTICAST (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MULTICAST (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 27]
```diff
@@ -846,9 +920,17 @@ g_inet_address_get_is_mc_global (GInetAddress *address)
   g_return_val_if_fail (G_IS_INET_ADDRESS (address), FALSE);
 
   if (address->priv->family == AF_INET)
-    return FALSE;
+    {
+      return FALSE;
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change.

File: gio/ginetaddress.c [hunk 28]
```diff
@@ -846,9 +920,17 @@ g_inet_address_get_is_mc_global (GInetAddress *address)
...
   else
-    return IN6_IS_ADDR_MC_GLOBAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MC_GLOBAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 29]
```diff
@@ -867,9 +949,17 @@ g_inet_address_get_is_mc_link_local (GInetAddress *address)
   g_return_val_if_fail (G_IS_INET_ADDRESS (address), FALSE);
 
   if (address->priv->family == AF_INET)
-    return FALSE;
+    {
+      return FALSE;
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change.

File: gio/ginetaddress.c [hunk 30]
```diff
@@ -867,9 +949,17 @@ g_inet_address_get_is_mc_link_local (GInetAddress *address)
...
   else
-    return IN6_IS_ADDR_MC_LINKLOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MC_LINKLOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 31]
```diff
@@ -888,9 +978,17 @@ g_inet_address_get_is_mc_node_local (GInetAddress *address)
   g_return_val_if_fail (G_IS_INET_ADDRESS (address), FALSE);
 
   if (address->priv->family == AF_INET)
-    return FALSE;
+    {
+      return FALSE;
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change.

File: gio/ginetaddress.c [hunk 32]
```diff
@@ -888,9 +978,17 @@ g_inet_address_get_is_mc_node_local (GInetAddress *address)
...
   else
-    return IN6_IS_ADDR_MC_NODELOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MC_NODELOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 33]
```diff
@@ -909,9 +1007,17 @@ g_inet_address_get_is_mc_org_local  (GInetAddress *address)
   g_return_val_if_fail (G_IS_INET_ADDRESS (address), FALSE);
 
   if (address->priv->family == AF_INET)
-    return FALSE;
+    {
+      return FALSE;
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change.

File: gio/ginetaddress.c [hunk 34]
```diff
@@ -909,9 +1007,17 @@ g_inet_address_get_is_mc_org_local  (GInetAddress *address)
...
   else
-    return IN6_IS_ADDR_MC_ORGLOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MC_ORGLOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: gio/ginetaddress.c [hunk 35]
```diff
@@ -930,9 +1036,17 @@ g_inet_address_get_is_mc_site_local (GInetAddress *address)
   g_return_val_if_fail (G_IS_INET_ADDRESS (address), FALSE);
 
   if (address->priv->family == AF_INET)
-    return FALSE;
+    {
+      return FALSE;
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Unnecessary change.

File: gio/ginetaddress.c [hunk 36]
```diff
@@ -930,9 +1036,17 @@ g_inet_address_get_is_mc_site_local (GInetAddress *address)
...
   else
-    return IN6_IS_ADDR_MC_SITELOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MC_SITELOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Block can be dropped to make the patch smaller.

File: glib/gnulib/vasnprintf.c [hunk 37]
```diff
@@ -91,6 +91,15 @@
 
 #include "verify.h"
 
+#ifdef __UCLIBC__
+int wctomb (char *__s, wchar_t __wchar) __THROW;
+
+int wctomb (char *__s, wchar_t __wchar)
```
Comment (oleavr) [inline glib/gnulib/vasnprintf.c]:
```suggestion
int
wctomb (char *__s, wchar_t __wchar)
```

**Review by oleavr — CHANGES_REQUESTED** [review]

File: gio/ginetaddress.c (hunk 16 above)
Comment (oleavr) [reply gio/ginetaddress.c]: Normally we would use a block, but it will make this patch larger and harder to maintain. (In case upstream isn't interested in this improvement.)

**Comment by WorksButNotTested** [comment]
Changes made.

File: gio/ginetaddress.c [hunk 38]
```diff
@@ -869,7 +933,13 @@ g_inet_address_get_is_mc_link_local (GInetAddress *address)
...
     return FALSE;
   else
-    return IN6_IS_ADDR_MC_LINKLOCAL (&address->priv->addr.ipv6);
+    {
+#ifdef HAVE_IPV6
+      return IN6_IS_ADDR_MC_LINKLOCAL (&address->priv->addr.ipv6);
+#else
+      return FALSE;
+#endif
+    }
```
Comment (oleavr) [inline gio/ginetaddress.c]: Only this guy left.

**Comment by WorksButNotTested** [comment]
Last one done.
